components/camel-sip/src/main/java/org/apache/camel/component/sip/SipConfiguration.java

```diff
@@ -53,124 +53,348 @@
-    @UriParam(label = "common")
-    private String eventId;
+
+    /**
+     * The amount of times a SIP message is allowed to be forwarded. Defaults to 70.
+     */
+    @UriParam(label = "proxy", defaultValue = "70")
+    private int maxForwards = 70;
+
+    /**
+     * The amount of time a message received at an endpoint is considered valid.
+     */
     @UriParam(label = "common", defaultValue = "3600")
     private int msgExpiration = 3600;
+
+    /**
+     * Determines whether requests are send via proxies.
+     */
     @UriParam(label = "proxy")
     private boolean useRouterForAllUris;
+
+    /**
+     * The amount of time to wait for a Response and/or Acknowledgement message to be received
+     * from another SIP stack. Defaults to 10 seconds.
+     */
     @UriParam(label = "common", defaultValue = "10000")
     private long receiveTimeoutMillis = 10000;
+
+    /**
+     * The maximum size of a message in bytes. Defaults to 1,048,576 B, roughly 1.05 MB.
+     */
     @UriParam(label = "advanced", defaultValue = "1048576")
     private int maxMessageSize = 1048576;
+
+    /**
+     * Whether connections should be cashed by the SipStack. This if useful for long
+     * running conversations as cashing reduces the cost of connection creations.
+     */
     @UriParam(label = "common")
-    private boolean cacheConnections;
+    private boolean cacheConnections; //todo check if this is used anywhere
+
+    /**
+     * The mime type of the body of the SIP message. Defaults to text.
+     */
     @UriParam(label = "common", defaultValue = "text")
     private String contentType = "text";
+
+    /**
+     * The mime subtype used in the body of the SIP message. Defaults to plain.
+     */
     @UriParam(label = "common", defaultValue = "plain")
     private String contentSubType = "plain";
+
+    /**
+     * Name of server log file to use for logging.
+     */
     @UriParam(label = "logging")
     private String implementationServerLogFile;
+
+    /**
+     * Name of client debug log file to use for logging.
+     */
     @UriParam(label = "logging")
     private String implementationDebugLogFile;
+
+    /**
+     * Logging level for tracing. Defaults to 0.
+     */
     @UriParam(label = "logging", defaultValue = "0")
     private String implementationTraceLevel = "0";
+
+    /**
+     * Whether the SipEndpoint is a consuming or producing endpoint. Determines if
+     * From headers are used (and thus the Endpoint is consuming because it <b>retrieves from</b> the SIP URI
+     * or
+     * To headers are used (and thus the Endpoint is producing because it needs to <b>send to</b> the SIP URI.
+     */
+    @UriParam(label = "consumer")
+    private boolean consumer;
+
+    /**
+     * Whether the sip consumer will subscribe to the given address or only listen for incoming
+     * MESSAGE requests
+     */
+    @UriParam(label = "consumer", defaultValue = "false")
+    private boolean subscribing = false;
```

Comment:
@nikvaessen is this default value consistent with the existing behavior of the component?